daemon/src/ipc_server.rs

```diff
@@ -146,6 +146,18 @@
             }
             IpcResponse::Ok
         }),
+
+        IpcMessage::SetWallpaper { monitor, wallpaper } => {
+            dbg!("SetWallpaper");
+            let monitors = vec![monitor];
+            check_monitors(wpaperd, &monitors).map(|_| {
+                for surf in collect_surfaces(wpaperd, monitors).iter_mut() {
+                    surf.set_new_wallpaper_path(wallpaper.as_path());
+                    surf.draw(&qh, 1_000).unwrap(); //TODO: should draw take an Option<'time'>?
```

Comment:
Yes, time should definitely be wrapped in an `Option<>`. However, `queue_draw` should be called in place of `draw`, as the former start loading the wallpaper in background until the next frame is drawn by the compositor. `draw` will basically try to run, it will not find the image loaded yet and behave almost the same as `queue_draw`, just with more overhead.

Reply:
Yeah, i left this on draft as these were still there.

Reply:
So `surf.queue_draw()` than `.draw()` ?
Apologies for the questions I'm having a bit of trouble parsing how all this comes together.

Reply:
No apologies needed! wpaperd has a background image loader, which means that the loading and decoding of an image will be done by separate threads. This is also important because the main thread is used for drawing the image to the screen and receive wayland events, so blocking here would not be good.

`surf.queue_draw()` just ask the background image loader to start loading the image (according to the path and so on) and then schedule wpaperd to run again when wayland draws another frame, until the image is loaded successfully. Running `draw()` at this point does the same thing, because the moment where we look for the loaded image, it's not there and the background image loader starts a new thread. The main difference is that we also do some GL switching and a couple of stuff that is not needed, and thus can be avoided in this case.

Reply:
Would you mind taking a look at whatever it is I am missing? (Following the calls etc I _thought_ that I'd plumbed everything in to where it is supposed to go -- but It looks like I'm missing something, as the current version is panicking when I try to run it.) [I'll respond on the other threads to keep things separate]